### ISSUE TITLE
Envoy 1.29.9 build can't download dependency

### ISSUE THREAD

**Issue by siepkes** [envoyproxy/envoy#36516]
The `googleurl` dependency in Envoy `1.29.9` can't be downloaded during build: https://github.com/envoyproxy/envoy/blob/94aa5f7f82fb543e7fbc011ea398ac12cc396817/bazel/repository_locations.bzl#L1188-L1202

Browsing to https://storage.googleapis.com/quiche-envoy-integration/dd4080fec0b443296c0ed0036e1e776df8813aa7.tar.gz gives an error like:

> AccessDeniedAccess denied.Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object. Permission 'storage.objects.get' denied on resource (or it may not exist).

Oddly enough I've also seen a different error indicating the bucket didn't exist. I didn't see any issues on the Google cloud status dashboard.

**Comment by nemacysts** [comment]
looks like this might be related to (well, fixed by) https://github.com/envoyproxy/envoy/pull/36515?

**Comment by siepkes** [comment]
@nemacysts Hm yeah good find, missed that one. 

The download now works but the build fails on the hash of the file being different. There is some discussion in https://github.com/envoyproxy/envoy/pull/36515 about hashes being different which might be what I'm seeing now (see error below).

@alyssawilk Subtle ping. I think this might be related to https://github.com/envoyproxy/envoy/pull/36515 ?

```
Starting local Bazel server and connecting to it...
INFO: Repository com_googlesource_googleurl instantiated at:
  /root/envoy-smartos/WORKSPACE:13:19: in <toplevel>
  /root/envoy-smartos/bazel/repositories.bzl:331:32: in envoy_dependencies
  /root/envoy-smartos/bazel/repositories.bzl:1141:26: in _com_googlesource_googleurl
  /root/envoy-smartos/bazel/repositories.bzl:54:23: in external_http_archive
  /root/.cache/bazel/_bazel_root/7558a64af10a6eb79f74e70211660103/external/envoy_api/bazel/envoy_http_archive.bzl:16:17: in envoy_http_archive
Repository rule http_archive defined at:
  /root/.cache/bazel/_bazel_root/7558a64af10a6eb79f74e70211660103/external/bazel_tools/tools/build_defs/repo/http.bzl:372:31: in <toplevel>
WARNING: Download from https://storage.googleapis.com/quiche-envoy-integration/dd4080fec0b443296c0ed0036e1e776df8813aa7.tar.gz failed: class com.google.devtools.build.lib.bazel.repository.downloader.UnrecoverableHttpException Checksum was fc694942e8a7491dcc1dde1bddf48a31370a1f46fef862bc17acf07c34dc6325 but wanted 59f14d4fb373083b9dc8d389f16bbb817b5f936d1d436aa67e16eb6936028a51
ERROR: An error occurred during the fetch of repository 'com_googlesource_googleurl':
   Traceback (most recent call last):
        File "/root/.cache/bazel/_bazel_root/7558a64af10a6eb79f74e70211660103/external/bazel_tools/tools/build_defs/repo/http.bzl", line 132, column 45, in _http_archive_impl
                download_info = ctx.download_and_extract(
Error in download_and_extract: java.io.IOException: Error downloading [https://storage.googleapis.com/quiche-envoy-integration/dd4080fec0b443296c0ed0036e1e776df8813aa7.tar.gz] to /root/.cache/bazel/_bazel_root/7558a64af10a6eb79f74e70211660103/external/com_googlesource_googleurl/temp14166583214422552352/dd4080fec0b443296c0ed0036e1e776df8813aa7.tar.gz: Checksum was fc694942e8a7491dcc1dde1bddf48a31370a1f46fef862bc17acf07c34dc6325 but wanted 59f14d4fb373083b9dc8d389f16bbb817b5f936d1d436aa67e16eb6936028a51
ERROR: /root/envoy-smartos/WORKSPACE:13:19: fetching http_archive rule //external:com_googlesource_googleurl: Traceback (most recent call last):
        File "/root/.cache/bazel/_bazel_root/7558a64af10a6eb79f74e70211660103/external/bazel_tools/tools/build_defs/repo/http.bzl", line 132, column 45, in _http_archive_impl
                download_info = ctx.download_and_extract(
Error in download_and_extract: java.io.IOException: Error downloading [https://storage.googleapis.com/quiche-envoy-integration/dd4080fec0b443296c0ed0036e1e776df8813aa7.tar.gz] to /root/.cache/bazel/_bazel_root/7558a64af10a6eb79f74e70211660103/external/com_googlesource_googleurl/temp14166583214422552352/dd4080fec0b443296c0ed0036e1e776df8813aa7.tar.gz: Checksum was fc694942e8a7491dcc1dde1bddf48a31370a1f46fef862bc17acf07c34dc6325 but wanted 59f14d4fb373083b9dc8d389f16bbb817b5f936d1d436aa67e16eb6936028a51
ERROR: @com_googlesource_googleurl//build_config:system_icu :: Error loading option @com_googlesource_googleurl//build_config:system_icu: java.io.IOException: Error downloading [https://storage.googleapis.com/quiche-envoy-integration/dd4080fec0b443296c0ed0036e1e776df8813aa7.tar.gz] to /root/.cache/bazel/_bazel_root/7558a64af10a6eb79f74e70211660103/external/com_googlesource_googleurl/temp14166583214422552352/dd4080fec0b443296c0ed0036e1e776df8813aa7.tar.gz: Checksum was fc694942e8a7491dcc1dde1bddf48a31370a1f46fef862bc17acf07c34dc6325 but wanted 59f14d4fb373083b9dc8d389f16bbb817b5f936d1d436aa67e16eb6936028a51
Loading: 
```

**Comment by phlax** [comment]
looks like this works now - at least where it has been fixed - on `main` and `release/v1.29` (see https://github.com/envoyproxy/envoy/commits/release/v1.29)

afaict the sha concern was a non issue - can we close this @siepkes ?

**Comment by siepkes** [comment]
Yes, 8e55b6b2525627dd2276c962b2de7bc5ab94c5c0 seems to fix the issue. Thanks everyone!
